src/shogun/preprocessor/PruneVarSubMean.cpp

```diff
@@ -102,7 +102,7 @@ void CPruneVarSubMean::cleanup()
 SGMatrix<float64_t>
 CPruneVarSubMean::apply_to_matrix(SGMatrix<float64_t> matrix)
 {
-	ASSERT(m_initialized)
+	REQUIRE(m_initialized, "Transformer has not been fitted.\n");
```

Comment:
offtopic: I would love to move all those general checks to base classes so that they dont get repeated in every specialization....
any thoughts on this

Reply:
shouldn't we use a different exception type other than ShogunException